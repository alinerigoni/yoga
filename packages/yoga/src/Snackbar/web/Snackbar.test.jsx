import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';

import { ThemeProvider, Snackbar } from '../..';

describe('<Snackbar />', () => {
  jest.useRealTimers();

  it('should match snapshot', () => {
    const { container } = render(
      <ThemeProvider>
        <Snackbar open message="Make wellbeing universal" />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render a minimal snackbar', () => {
    render(
      <ThemeProvider>
        <Snackbar open message="Make wellbeing universal" />
      </ThemeProvider>,
    );

    screen.getByRole('alert');
    screen.getByRole('img');
    screen.getByLabelText('success');
    screen.getByText('Make wellbeing universal');
  });

  it('should not render an icon', () => {
    render(
      <ThemeProvider>
        <Snackbar open message="Make wellbeing universal" hideIcon />
      </ThemeProvider>,
    );

    expect(screen.queryByRole('img')).toBeNull();
  });

  it('should render a custom action button', () => {
    const onActionMock = jest.fn();

    render(
      <ThemeProvider>
        <Snackbar
          open
          message="Make wellbeing universal"
          onAction={onActionMock}
          actionLabel="Let's go"
        />
      </ThemeProvider>,
    );

    const button = screen.getByRole('button', { name: /let's go/i });

    fireEvent.click(button);

    expect(onActionMock).toHaveBeenCalledTimes(1);
  });

  it('should render a close button', () => {
    const onCloseMock = jest.fn();

    render(
      <ThemeProvider>
        <Snackbar
          open
          message="Make wellbeing universal"
          onClose={onCloseMock}
        />
      </ThemeProvider>,
    );

    const button = screen.getByRole('button');

    fireEvent.click(button);

    expect(onCloseMock).toHaveBeenCalledTimes(1);
  });

  it('should close after timeout', async () => {
    jest.useFakeTimers();

    const onCloseMock = jest.fn();

    render(
      <ThemeProvider>
        <Snackbar
          open
          duration={1000}
          message="Make wellbeing universal"
          onClose={onCloseMock}
        />
      </ThemeProvider>,
    );

    expect(screen.queryByRole('button')).toBeNull();

    jest.runAllTimers();

    expect(onCloseMock).toHaveBeenCalledTimes(1);
  });
});
