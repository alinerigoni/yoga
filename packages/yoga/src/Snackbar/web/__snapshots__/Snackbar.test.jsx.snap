// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Snackbar /> should match snapshot 1`] = `
.c3 {
  margin: 0;
  padding: 0;
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  font-family: Rubik;
  color: #231B22;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 2;
  overflow: hidden;
}

.c1 {
  margin-right: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  width: 24px;
  height: 24px;
}

.c4 {
  margin-right: 16px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 svg {
  margin-left: 16px;
  margin-right: 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: absolute;
  bottom: 16px;
  right: 16px;
  left: 16px;
  min-width: 320px;
  max-width: 600px;
  min-height: 56px;
  max-height: 72px;
  padding: 16px;
  border-radius: 8px;
  box-shadow: 0 2px 4px -1px rgba(152,152,166,0.2),0 4px 5px 0px rgba(152,152,166,0.14),0 1px 10px 0px rgba(152,152,166,0.12);
  background-color: #C1EEDB;
  z-index: 100;
  -webkit-animation: bcCCNc 0.2s ease-in-out;
  animation: bcCCNc 0.2s ease-in-out;
}

@media (min-width:768px) {
  .c0 {
    left: auto;
    bottom: 40px;
    right: 40px;
    min-width: 400px;
  }
}

<div>
  <div
    aria-label="success"
    class="c0"
    role="alert"
  >
    <div
      class="c1"
      display="flex"
      role="img"
    >
      <svg
        class="c2"
        fill="#231B22"
        height="24"
        width="24"
      />
    </div>
    <p
      class="c3 c4"
    >
      Make wellbeing universal
    </p>
    <aside
      class="c5"
    />
  </div>
</div>
`;
